cli/azd/pkg/commands/builder.go

```diff
@@ -24,12 +28,31 @@ func Build(action Action, rootOptions *GlobalCommandOptions, use string, short s
 
 			// Set the global options in the go context
 			ctx = azdcontext.WithAzdContext(ctx, azdCtx)
-			ctx = WithGlobalCommandOptions(ctx, rootOptions)
+			ctx = internal.WithCommandOptions(ctx, *rootOptions)
+
+			azCliArgs := azcli.NewAzCliArgs{
+				EnableDebug:     rootOptions.EnableDebugLogging,
+				EnableTelemetry: rootOptions.EnableTelemetry,
+			}
 
 			// Create and set the AzCli that will be used for the command
-			azCli := GetAzCliFromContext(ctx)
+			azCli := azcli.NewAzCli(azCliArgs)
 			ctx = azcli.WithAzCli(ctx, azCli)
 
+			formatter, err := output.GetCommandFormatter(cmd)
+			if err != nil {
+				log.Printf("getting formatter: %s", err.Error())
+				formatter, _ = output.NewFormatter(string(output.NoneFormat))
+			}
+
+			ctx = output.WithFormatter(ctx, formatter)
```

Comment:
I feel like this will cause excessive error checking all over the place wherever we get a value from the context.  I think returning an intelligent default implementation would be a better developer experience except for the use cases where a default doesn't make sense.